aci/resource_aci_rtctrlattrp.go

```diff
@@ -128,6 +258,62 @@ func resourceAciActionRuleProfileCreate(ctx context.Context, d *schema.ResourceD
 		return diag.FromErr(err)
 	}
 
+	// rtctrlSetComm - Operations
+	if setComm, ok := d.GetOk("set_communities"); ok {
+
+		log.Printf("[DEBUG] rtctrlSetComm: Beginning Creation")
+		rtctrlSetCommAttr := models.RtctrlSetCommAttributes{}
+
+		setCommMap := toStrMap(setComm.(map[string]interface{}))
+		rtctrlSetCommAttr.Community = setCommMap["community"]
+		rtctrlSetCommAttr.SetCriteria = setCommMap["criteria"]
+		rtctrlSetComm := models.NewRtctrlSetComm(fmt.Sprintf(models.RnrtctrlSetComm), rtctrlAttrP.DistinguishedName, "", "", rtctrlSetCommAttr)
+
+		creation_err := aciClient.Save(rtctrlSetComm)
+		if creation_err != nil {
+			return diag.FromErr(creation_err)
+		}
+
+		log.Printf("[DEBUG] %s: Creation finished successfully", rtctrlSetComm.DistinguishedName)
+		resourceAciRtctrlSetCommRead(ctx, rtctrlSetComm.DistinguishedName, d, m)
```

Comment:
Why reading here and also in the main read function called at the end? Isn't this a double read ? (See same comment in other PR and this apply to all the reads for sub-options)

Reply:
As per the code, we developed a Read function call for all independent child classes and at the end of the Create or Update function we are calling the Action Rule Profile read function.